operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ExecutorServiceManager.java

```diff
@@ -68,6 +70,31 @@ public ExecutorService workflowExecutorService() {
     return workflowExecutor;
   }
 
+  /**
+   * Runs the specified I/O-bound task and waits for its completion using the ExecutorService
+   * provided by {@link #workflowExecutorService()}
+   *
+   * @param task task to run concurrently
+   */
+  public static void executeAndWaitForCompletion(Runnable task) {
+    executeAndWaitForCompletion(task, instance().workflowExecutorService());
```

Comment:
It is a little smelly that this workflow executor is reused for a general purpose executor service IMO.